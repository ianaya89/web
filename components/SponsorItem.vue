<template>
  <div class="sponsor-item flex items-center">
    <div class="w-16">
      <avatar class="my-auto" :src="src" :rounded="false" />
    </div>
    <div class="w-3/5 ml-4">
      <h6 class="my-auto leading-tight">
        {{ name }}
      </h6>
    </div>
  </div>
</template>

<script>
import Avatar from '@/components/Avatar'

export default {
  name: 'SponsorItem',

  components: { Avatar },

  props: {
    src: {
      type: String,
      default: require('../assets/images/thumb-fallback.svg')
    },
    name: {
      type: String,
      required: true
    }
  }
}
</script>

<style lang="postcss" scoped>
.sponsor-item {
  @apply .w-full .flex .border-b .border-grey-l4 .p-6;
  transition: border-color 400ms linear;
  will-change: border-color;

  &:hover {
    @apply .border-red;
  }
}
</style>
