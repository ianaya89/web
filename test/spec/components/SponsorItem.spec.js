import { shallowMount } from '@vue/test-utils'
import SponsorItem from '@/components/SponsorItem'

describe('SponsorItem', () => {
  it('renders SponsorItem with name', () => {
    const wrapper = shallowMount(SponsorItem, {
      propsData: {
        name: 'John'
      }
    })

    expect(wrapper.element).toMatchSnapshot()
  })
})
