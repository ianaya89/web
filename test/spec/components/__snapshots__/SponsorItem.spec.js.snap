// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SponsorItem renders SponsorItem with name 1`] = `
<div
  class="sponsor-item flex items-center"
>
  <div
    class="w-16"
  >
    <avatar-stub
      avatarclasses=""
      class="my-auto"
      classes="h-16 w-16"
      src="\\"thumb-fallback.svg\\""
      tooltipcolor="light"
      tooltipposition="top"
    />
  </div>
   
  <div
    class="w-3/5 ml-4"
  >
    <h6
      class="my-auto leading-tight"
    >
      
      John
    
    </h6>
  </div>
</div>
`;
